src/Elliptic/Systems/Poisson/Observe.hpp

```diff
@@ -0,0 +1,124 @@
+#include "Utilities/Numeric.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Poisson {
+namespace Actions {
+
+/*!
+ * \brief Temporary action to observe volume and reduction data for the Poisson
+ * system
+ */
+struct Observe {
+  template <typename... DbTags, typename... InboxTags, typename Metavariables,
+            size_t Dim, typename ActionList, typename ParallelComponent>
+  static auto apply(db::DataBox<tmpl::list<DbTags...>>& box,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ElementIndex<Dim>& array_index,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    const auto& iteration_id = get<LinearSolver::Tags::IterationId>(box);
+    const auto& mesh = get<Tags::Mesh<Dim>>(box);
+    const std::string element_name = MakeString{} << ElementId<Dim>(array_index)
+                                                  << '/';
+
+    // Retrieve the current numeric solution
+    const auto& field = get<Poisson::Field>(box);
+
+    // Compute the analytic solution
+    const auto& inertial_coordinates =
+        db::get<::Tags::Coordinates<Dim, Frame::Inertial>>(box);
+    const auto field_analytic = get<Poisson::Field>(
+        Parallel::get<typename Metavariables::analytic_solution_tag>(cache)
+            .field_variables(inertial_coordinates));
+
+    // Compute error between numeric and analytic solutions
+    DataVector field_error = get(field) - get(field_analytic);
+
+    // Retrieve linear residual
+    // We have to reduce it here again, even though it should be the same on all
+    // elements, because it was computed in the linear solver residual monitor
+    // singleton. Ideally we would contribute it to the observation from the
+    // singleton when that becomes possible.
+    using fields_tag = typename Metavariables::system::fields_tag;
+    using residual_magnitude_tag = db::add_tag_prefix<
+        LinearSolver::Tags::Magnitude,
+        db::add_tag_prefix<LinearSolver::Tags::Residual, fields_tag>>;
+    const double linear_residual = get<residual_magnitude_tag>(box);
+
+    // Compute l2 error square over local element
+    const double local_l2_error_square =
+        definite_integral(square(field_error), mesh);
+
+    // Collect volume data
+    // Remove tensor types, only storing individual components
+    std::vector<TensorComponent> components;
+    components.reserve(3 + Dim);
+    components.emplace_back(element_name + Poisson::Field::name(), get(field));
+    components.emplace_back(element_name + Poisson::Field::name() + "Analytic",
+                            get(field_analytic));
+    components.emplace_back(element_name + Poisson::Field::name() + "Error",
+                            field_error);
+    components.emplace_back(element_name + "InertialCoordinates_x",
```

Comment:
Should this be either of "inertial_coordinates_x" or "InertialCoordinatesX"? I'm unsure if we have a style convention for names in strings, but since file names are CamelCased I would also CamelCase this.

Reply:
This naming convention is used in the other observers. I think the idea is to have the label in CamelCase and append the tensor component (not part of the label) with an underscore.

Reply:
Thank you for the clarification :+1: I will keep this in mind along with the other naming conventions listed in the code review guidelines.